callback-query.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"fmt"
+	"github.com/go-telegram-bot-api/telegram-bot-api"
+	"log"
+	"regexp"
+	"strings"
+)
+
+func answerCallbackWithNotification(x *opBot, callbackID, text string) error {
+	_, err := x.bot.AnswerCallbackQuery(
+		tgbotapi.CallbackConfig{
+			CallbackQueryID: callbackID,
+			Text:            text,
+			ShowAlert:       true,
+		},
+	)
+	return err
+}
+
+func extractRequestID(data, prefix, errorMsg string) (string, error) {
+	regex := regexp.MustCompile(fmt.Sprintf(`%s-(\S+)`, prefix))
+	matches := regex.FindStringSubmatch(data)
+	if matches == nil {
+		err := fmt.Errorf("%s: %q", errorMsg, data)
+		log.Printf("extractRequestID: %v", err)
+		return "", err
+	}
+	return matches[1], nil
+}
+
+func handleCallbackQuery(x *opBot, update tgbotapi.Update) error {
+	data := update.CallbackQuery.Data
+
+	switch {
+	case data == "rules":
+		answerCallbackWithNotification(x, update.CallbackQuery.ID, T("rules"))
+	case strings.HasPrefix(data, "ban-user-"):
+		requestID, err := extractRequestID(data, "ban-user", "malformed ban request callback query")
+		if err != nil {
+			answerCallbackWithNotification(x, update.CallbackQuery.ID, T("callback_invalid_request"))
+			break
+		}
+		// We pass `true' as parameter to indicate we want to ban the user as well.
+		if deleteMessageFromBanRequest(x, update.CallbackQuery.From, requestID, true) != nil {
```

Comment:
Maybe set a default message and override in case of error? This would allow only one answerCallbackWithNotification after this block.